crates/turborepo-cache/src/cache_archive/create.rs

```diff
@@ -178,4 +402,21 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_compression() -> Result<()> {
+        let mut buffer = Vec::new();
+        let mut encoder = zstd::Encoder::new(&mut buffer, 0)?.auto_finish();
+        encoder.write(b"hello world")?;
+        // Should finish encoding on drop
+        drop(encoder);
```

Comment:
That feels odd to me but it makes sense. Coming from the hash / digest api I am surprised but I guess since the digest apis spit out data when you finalize rather than (I assume) write it as you go it makes sense to have 'implicit finalize' on drop.

Reply:
Yeah we could require an explicit finalize, but that's too easy to miss imo, so it's probably better to just finalize on drop